parquet-tools/src/main/java/org/apache/parquet/tools/command/CatCommand.java

```diff
@@ -71,14 +75,19 @@ public void execute(CommandLine options) throws Exception {
     try {
       PrintWriter writer = new PrintWriter(Main.out, true);
       reader = ParquetReader.builder(new SimpleReadSupport(), new Path(input)).build();
+      ParquetMetadata metadata = ParquetFileReader.readFooter(new Configuration(), new Path(input));
+      JsonRecordFormatter.JsonGroupFormatter formatter = JsonRecordFormatter.fromSchema(metadata.getFileMetaData().getSchema());
+
       for (SimpleRecord value = reader.read(); value != null; value = reader.read()) {
         if (options.hasOption('j')) {
-          value.prettyPrintJson(writer);
+          writer.write(formatter.formatRecord(value));
```

Comment:
Reading you comment again I think I misunderstood your question. Basically, the issue is with `prettyPrintJson`:

```
protected Object toJsonObject() {
    Map<String, Object> result = Maps.newLinkedHashMap();
    for (NameValue value : values) {
      // This doesn't work because there might be  multiple values with the same key - OVERWRITE.
      // Could change this to a Map<String, Collection> but then we have the same problem in reverse
      // that is, most entries will have only one entry while other items may be arrays. We could
      // just print single-item collections as non-arrays, and multiple item collections as arrays
      // but that's very dirty. Doing this sort of formatting here breaks the 'Single Responsibility Principle' (TM) anyway. 
      result.put(value.getName(), toJsonValue(value.getValue()));
    }

    return result;
  }

  protected static Object toJsonValue(Object val) {
    if (SimpleRecord.class.isAssignableFrom(val.getClass())) {
      return ((SimpleRecord) val).toJsonObject();
    } else if (byte[].class == val.getClass()) {
      return new BinaryNode((byte[]) val);
    } else {
      return val;
    }
  }
``` 

It's the same basic issue I outlined in my previous comment (but I was addressing the design choice rather than why it was failing in the first place - which was your actual question I think). Please see the comment above.

Thanks